### ISSUE TITLE
Collected the ads, stuck at "Pending"

### ISSUE THREAD

**Issue by MrDemocracy** [dhowe/AdNauseam#1098]
### Describe the issue
When AdNauseam collects ads from a page, it just says "Pending" on all the items. Nothing happens.
When I reinstall AdNauseam, it successfully collects and click ads for a short while, and suddenly I'm stuck at "Pending" again.

### One or more specific URLs where the issue occurs
Not relevant.

### Screenshot in which the issue can be seen
<img width="400"  src="https://user-images.githubusercontent.com/23060864/28436879-dbaef162-6d98-11e7-8c37-47d5d7c17e9c.png">

### Steps for anyone to reproduce the issue
(Please be as detailed as possible)
Don't know how to reproduce this. I'm able to manually open the ads by clicking on one of the pending ones.
### Your settings

- OS/version: macOS Sierra 10.12.6 beta 6 (16G24b)
- Browser/version: Google Chrome Version 59.0.3071.115 (Official Build) (64-bit)
- AdNauseam version: v3.3.403
- Other extensions you have installed:

<details> 
  <summary>List of extensions I have installed </summary>
<tb>9gag GIF Downloader 0.1</tb>

<tb>9gag Night Mode 1.2.2</tb>
Night mode for the 9gag website!
9gif 1.1
AliExpress Tools 3.8.4
AliTools — your reliable assistant on AliExpress
AllCast Receiver 1.0.0.8
App Runtime for Chrome (Beta) 54.5021.629.0
AppleTV AirPlay Remote 0.3
Application Launcher for Drive (by Google) 3.2
Apps Launcher 2.4.9
ARC Welder 54.5021.629.0
CCTV Monitor 1.2.1
Chrome Currency Converter 4.3.23
Chrome Remote Desktop 60.0.3112.25
Chrono Download Manager 0.9.3
CIRC 0.6.7
ColorZilla 2.0
cookies.txt 1.10.1
CouchPotato 1.2.2
eBay for Chrome 5.2.2
EditThisCookie 1.4.1
Emoji for Google Chrome™ 3.3.2
F5 Shopping Master - Best Price - Save Money 4.1
Fatkun Batch Download Image 2.11
Flash Video Downloader 28.1.1
Frameless for Netflix 0.2.1
External Frameless Window for Netflix
Free Proxy to Unblock any sites | Touch VPN 1.5.12
Google Cast for Education 17.630.1.3
Google Docs 0.9
Google Docs Offline 1.4
Google Hangouts 2017.420.419.1
Google Keep - notes and lists 3.1.17284.467
Google Keep Chrome Extension 3.1.16302.1110
Google Sheets 1.1
Google Slides 0.9
GoogleGIFs 1.1
Grammarly for Chrome 14.791.1063
HTTPS Everywhere 2017.7.18
Imagus 0.9.8.54
IP Domain Country Flag 0.1.6
J2TeaM - 9GAG Tools 1.0.0
mobile browser emulator 1.3.2
MusicMap 0.2.1
Musixmatch Lyrics for YouTube 2.2.1
Native HLS Playback 0.7.10
Netflix Pause Removal 1.1.0
Open-as-Popup 1.0.0.1
Password Alert 1.25
Plex 2.12.9
Pop up blocker for Chrome™ - Poper Blocker 4.0.5.5
Popup Blocker Pro 1.3.5
Pushbullet 336
Reddit Enhancement Suite 5.8.2
Reddit Notifications 0.4.1
Send Anywhere (File Transfer) 7.2.10
Send Anywhere(File Transfer, for Gmail/Slack) 7.7.13
Session Buddy 3.6.2
sFTP Client 2.5.1.1
skyZIP™ Proxy 0.8.1
Speedtest by Ookla 1.0.8
Streaming Media Player 7.6
Stylebot 2.2
Super Netflix 2.1.0.1028
Super Simple Highlighter 3.10.1
Tampermonkey 4.3.6
The Camelizer 2.9.2
Traktflix - Netflix and Trakt.tv together 1.0.8.1
User-Agent Switcher 2.0
vGet Cast (DLNA Controller) 0.1.11
vGet Extension (Video Downloader, DLNA) 0.1.11
Video Downloader GetThemAll 30.0.1
Video Downloader professional 1.97.54
Video Player 2.0.10
Videostream for Google Chromecast™ 2.17.501.0
VNC® Viewer for Google Chrome™ 1.2.2.15132
Vysor 1.8.0
Webmail Ad Blocker 5.0.4.1
Will It Stream 1.0
Word Replacer II 2.0.10
ZenMate VPN - Best Cyber Security & Unblock 5.10.4
 </details>

**Comment by dhowe** [comment]
So ads are collected, but not visited -- can you include a screenshot of your settings page? Thanks...

**Comment by MrDemocracy** [comment]
<img width="1280"  src="https://user-images.githubusercontent.com/23060864/28454016-eecd51c4-6df9-11e7-8a45-97a9cbd3d529.png">

**Comment by MrDemocracy** [comment]
Could you also please read my [reddit post](https://www.reddit.com/r/Adblock/comments/6ofdy7/adnauseam_fails_to_click_ad_on_fmoviesis/), for another issue I have with AdNauseam?

**Comment by MrDemocracy** [comment]
Two other things that would be nice if you could add, is filter (search feature) for the AdVault, and when you hover over "clear" button, three new options will appear: "Clear Pending", "Clear Failed", "Clear Pending & Failed".

**Comment by MrDemocracy** [comment]
How long is it before a pending item timeouts? Is error 404 treated as "failed", or do all items follow the timeout period?

**Comment by dhowe** [comment]
Each ad is tried 3 times before being marked as 'failed'
404 error counts as one failed attempt
Timeout is quite conservative (20 sec)

> Two other things that would be nice if you could add, is filter (search feature) for the AdVault, and when you hover over the "clear" button, three new options will appear: "Clear Pending", "Clear Failed", "Clear Pending & Failed".
> Edit: Another thing, a toggle in settings for "Auto-Clear Failed Ads".

These are items that should be part of advault filtering, which actually needs some reworking (see )...

**Comment by MrDemocracy** [comment]
This should also be implemented:
<img width="1280" src="https://user-images.githubusercontent.com/23060864/28582671-87e3ec8c-7166-11e7-963a-94fc7043ae51.png">

**Comment by inglorious4** [comment]
Same problem that MrDemocracy is experiencing. Ads are collected and listed at "pending" state, but never get clicked. Running Windows 10 64 bit, Opera 64 bit with version 3.3.403 AdNauseam. Visited the following page - http://thefreethoughtproject.com/nurse-not-alone-police-history-abusing-professionals/ - checked the ads on the page and noticed all collected ads were at "pending" status. Waited 10 minutes then closed the browser. Re-opened and visited the page again, ads were collected and clicked almost immediately. Adnauseam settings:
![adn_settings](https://user-images.githubusercontent.com/31597850/30006269-9d7ef228-90c2-11e7-9351-efb01b50c017.JPG)

Only other plugins installed / active:
Lastpass
Zoomba
Open In Background Tab

Also of note - running on a laptop. I never shut the laptop down I simply close the lid to put it into sleep mode. I'm trying to determine if the laptop going into sleep mode and being woken up might be contributing to the issue. Will update accordingly.

**Comment by Emptylillie** [comment]
Same problem as other posters. After 10-20 minutes of browsing AdNauseam stops clicking on ads and has them 'pending' forever. Restarting a browser fixes this problem, but after a while it appears again.
![2017-09-10 23_29_14- opera](https://user-images.githubusercontent.com/31829748/30252701-f2a4a7bc-967f-11e7-9e32-fbd433e70dde.jpg)

OS: Win 7 x64
Browser: Opera 47.0.2631.80
AdNauseam v3.3.403
AdNauseam settings
![2017-09-10 23_10_58-adnauseam dashboard - opera](https://user-images.githubusercontent.com/31829748/30252629-c33e2eea-967e-11e7-99cf-5b642996ddc9.jpg)

Other extensions:
Don't track me google
Download Chrome extention
Reddit Enhancement Suite
SoundCloud Scrobbler
Stylish

**Comment by dhowe** [comment]
It may the case that these ads simply have not been clicked _yet_
Ads are clicked on a randomized timer so that they don't all happen within too short of a time window
So it is possible that other ads are being clicked first
A better way to test this is to open the advault, then refresh the page after a few seconds and see if the 'ads clicked' count is changing
Or, you can also 'activate debugging mode' in the settings, then check the addon console

**Comment by dhowe** [comment]
Question: do you see still see pending after you close and re-open the AdNauseam popup menu?

**Comment by inglorious4** [comment]
Sorry for the late response.  I actually uninstalled AdNauseam due to the pending issue and I have since moved on to use the Brave browser.  I do remember though that when I was using Opera and had AdNauseum installed I could re-open the popup menu many times and still see pending.  Seemed the only thing I could do was to kill the browser and restart it would then start working again.  At least for a short time before the pending issue started again.
